### PR TITLE
Make sure test runs are deterministic

### PR DESCRIPTION
**Pull request by ausi**
Same as #2879 for Contao 4.11

### CODE REVIEW

File: core-bundle/tests/Routing/RouteProviderTest.php [hunk 1]
```diff
@@ -29,6 +29,15 @@
 
 class RouteProviderTest extends TestCase
 {
+    private $pageModelAutoIncrement = 0;
+
+    protected function tearDown(): void
+    {
+        $this->pageModelAutoIncrement = 0;
+
+        parent::setUp();
```
Comment (aschempp) [inline core-bundle/tests/Routing/RouteProviderTest.php]: you sure?

File: core-bundle/tests/Security/TwoFactor/AuthenticatorTest.php [hunk 2]
```diff
@@ -126,4 +126,16 @@ public function testGeneratesTheQrCode(): void
         $this->assertSame(5897, \strlen($svg));
         $this->assertSame(0, strpos($svg, $beginSvg));
     }
+
+    private function randomBytes(int $bytesCount, int $seed): string
```
Comment (aschempp) [inline core-bundle/tests/Security/TwoFactor/AuthenticatorTest.php]: I guess you should also rename this method.

**Review by aschempp — CHANGES_REQUESTED** [review]

File: core-bundle/tests/Routing/RouteProviderTest.php (hunk 1 above)
Comment (ausi) [reply core-bundle/tests/Routing/RouteProviderTest.php]: Thats already fixed in #2879, I need to update this PR.

File: core-bundle/tests/Security/TwoFactor/AuthenticatorTest.php (hunk 2 above)
Comment (ausi) [reply core-bundle/tests/Security/TwoFactor/AuthenticatorTest.php]: Yes, need to integrate changes from #2879

File: core-bundle/tests/Routing/RouteProviderTest.php (hunk 1 above)
Comment (ausi) [reply core-bundle/tests/Routing/RouteProviderTest.php]: Done in 5271027181ae7f966ebd1e450fa45f3eb5cceba8

File: core-bundle/tests/Security/TwoFactor/AuthenticatorTest.php (hunk 2 above)
Comment (ausi) [reply core-bundle/tests/Security/TwoFactor/AuthenticatorTest.php]: Done in 5271027181ae7f966ebd1e450fa45f3eb5cceba8

**Review by aschempp — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**